Sources/Rendering/Core/ImageMapper/index.js

```diff
@@ -101,38 +213,80 @@ function vtkImageMapper(publicAPI, model) {
       return image.getBounds();
     }
 
-    const ex = [
-      model.customDisplayExtent[0],
-      model.customDisplayExtent[1],
-      model.customDisplayExtent[2],
-      model.customDisplayExtent[3],
-      model.zSlice,
-      model.zSlice,
-    ];
+    const ex = model.customDisplayExtent.slice();
+    let nSlice;
+    switch (model.currentSlicingMode) {
+      case SlicingMode.I:
+        ex[0] = model.slice;
+        ex[1] = model.slice;
+        break;
+      case SlicingMode.J:
+        ex[2] = model.slice;
+        ex[3] = model.slice;
+        break;
+      case SlicingMode.K:
+        ex[4] = model.slice;
+        ex[5] = model.slice;
+        break;
+      case SlicingMode.X:
+        nSlice = publicAPI.getSliceAtPosition(model.slice);
+        ex[0] = nSlice;
+        ex[1] = nSlice;
+        break;
+      case SlicingMode.Y:
+        nSlice = publicAPI.getSliceAtPosition(model.slice);
+        ex[2] = nSlice;
+        ex[3] = nSlice;
+        break;
+      case SlicingMode.Z:
+        nSlice = publicAPI.getSliceAtPosition(model.slice);
+        ex[4] = nSlice;
+        ex[5] = nSlice;
+        break;
+      default:
+        break;
+    }
 
     return image.extentToBounds(ex);
   };
 
-  publicAPI.getBoundsForSlice = (slice, thickness = 0) => {
+  publicAPI.getBoundsForSlice = (slice = model.slice, thickness = 0) => {
     const image = publicAPI.getInputData();
     if (!image) {
       return vtkMath.createUninitializedBounds();
     }
     const extent = image.getExtent();
-    switch (publicAPI.getCurrentSlicingMode()) {
-      case SlicingMode.X:
+    let nSlice;
+    switch (model.currentSlicingMode) {
+      case SlicingMode.I:
         extent[0] = slice - thickness;
         extent[1] = slice + thickness;
         break;
-      case SlicingMode.Y:
+      case SlicingMode.J:
         extent[2] = slice - thickness;
         extent[3] = slice + thickness;
         break;
-      case SlicingMode.Z:
+      case SlicingMode.K:
         extent[4] = slice - thickness;
         extent[5] = slice + thickness;
         break;
+      case SlicingMode.X:
+        nSlice = publicAPI.getSliceAtPosition(model.slice);
+        extent[0] = nSlice - thickness;
+        extent[1] = nSlice + thickness;
+        break;
+      case SlicingMode.Y:
+        nSlice = publicAPI.getSliceAtPosition(model.slice);
+        extent[2] = nSlice - thickness;
+        extent[3] = nSlice + thickness;
+        break;
+      case SlicingMode.Z:
+        nSlice = publicAPI.getSliceAtPosition(model.slice);
+        extent[4] = nSlice - thickness;
+        extent[5] = nSlice + thickness;
+        break;
```

Comment:
Do not know that X==i here etc

Reply:
Done:
- https://github.com/Kitware/vtk-js/pull/591/commits/8b80c6ff33c104f6be3c450f7cad186a26afc26b#diff-9762803aa1d1e7c8613c71d155ed4e0dR215
- https://github.com/Kitware/vtk-js/pull/591/commits/8b80c6ff33c104f6be3c450f7cad186a26afc26b#diff-9762803aa1d1e7c8613c71d155ed4e0dR247